src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -381,15 +381,20 @@ private static TableReference getTableReference(Configuration conf) throws IOExc
       return Optional.empty();
     }
 
-    private void operationAction(TableReference tableRef) throws InterruptedException {
+    private void operationAction(TableReference tableRef, @Nullable String cmekKey) throws InterruptedException {
       if (allowSchemaRelaxation) {
         updateTableSchema(tableRef);
       }
       String query = generateQuery(tableRef);
       LOG.debug("Update/Upsert query: " + query);
 
       BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
-      QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(query).setUseLegacySql(false).build();
+      QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(query)
+          .setUseLegacySql(false)
+          .setDestinationEncryptionConfiguration(
+            com.google.cloud.bigquery.EncryptionConfiguration.newBuilder().setKmsKeyName(cmekKey).build())
```

Comment:
is the full class name required?

Reply:
Yes, the load config uses a different `EncrptionConfiguration`, https://github.com/data-integrations/google-cloud/blob/77e98bc7f7d7630f2e8f113c986b140799458934/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java#L241